### DISCUSSION TITLE
How to automatically type cast query results of json column?

### DISCUSSION THREAD

**Discussion by paymog** [prisma/prisma#20771] [Q&A] [answered]
### Question

I have a table, `EventLog`, with a `Json` column, `payload`, and another enum column `EventType`. 

```
enum EventType {
  UsersInvited
}

model EventLog {
  event_type      EventType
  payload         Json
}
```

At the application layer I ensure that rows of the same `EventType` have the same shape for `payload`. For example, an event of type `UsersInvited` will always of a `payload` with a shape of `{id: string, name: string, birthday: string}[]`.

Does Prisma offer a way of automatically casting the `payload` of queries for `UserInvited` rows to `{id: string, name: string, birthday: string}[]`? Alternatively, is there an easy way to use composition/inheritance with/of the Prisma client to achieve this? 

For example, is there a clean way to simplifies the following query patterns?

```ts
const first = await prisma.eventLog.findFirst({ where: {event_type: EventType.UsersInvited } })
const context = first?.context as {id: string, name: string, birthday: string}[]

const many = await prisma.eventLog.findMany({ where: {event_type: EventType.UsersInvited } })
const manyContextCasted =  many.map((e) => {
        return { ...e, context: e.context as {id: string, name: string, birthday: string}[] };
      });

// same for findFirstOrThrow
```

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment, marked as answer]
Hi @paymog 👋 

Prisma does not currently offer a built-in way to automatically cast the payload of queries for specific rows to a certain type. For this, you might need to manually cast the payload in your application code as you're currently doing.

You can, however open a [feature request](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) and document your use case so that our engineers can take a look.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏
